flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/state/PendingSplitsStateSerializerTest.java

```diff
@@ -102,6 +104,22 @@ public void testRepeatedSerializationCache() throws Exception {
         assertSame(ser1, ser3);
     }
 
+    @Test
+    public void testOutputIsFinallyCleared() throws Exception {
+        final PendingSplitsStateSerializer serializer =
+                new PendingSplitsStateSerializer(MySqlSplitSerializer.INSTANCE);
+
+        final byte[] ser1 = serializer.serialize(state);
+        state.serializedFormCache = null;
+
+        PendingSplitsState unsupportedState = new UnsupportedPendingSplitsState();
+
+        assertThrows(IOException.class, () -> serializer.serialize(unsupportedState));
+
+        final byte[] ser2 = serializer.serialize(state);
+        assertEquals(ser1.length, ser2.length);
```

Comment:
Is it better to assert that `ser1` and `ser2` are the same, since there shouldn't be any changes to the state if an error happens based on my understanding.

Reply:
For some reason, they aren't.